0000-pando-cli-install.md

```diff
@@ -0,0 +1,104 @@
+- Start Date: 2023-08-10
+- RFC PR: (leave this empty)
+- Pando Issue: (leave this empty)
+
+# Before Reading
+
+Please approach this document with a generous perspective understanding that it
+is a Markdown file so all examples are _theory_ and **have not been prototyped**.
+You may see "gotchas" which is normal.
+
+If you choose to contribute, please address the overall purpose of the RFC and not
+nit-picking the "gotchas" in code examples. :heart:
+
+# Summary
+
+A cli ([command line interface](https://www.w3schools.com/whatis/whatis_cli.asp)) tool to easily scaffold a Pando project
+
+# Motivation
+
+One of the most difficult aspects of pattern adoption is the initial hurdle - getting folks to actually try it out. There could be a variety of reasons for this. Some possible scenarios: 
+1. Time constraints can make it difficult to try out a new tool if it does not have proven value
+1. Documentation can be difficult to parse
+1. Some people just learn better by doing than reading
+
+A cli would allow the user to enter a command into a terminal and immediately interact with the design system. This could be used as a tool to onboard and learn Pando, or include and use in an existing project.
+
+The expected outcome would be that there would be an increased adoption and usage of Pando. This could lead to more positive feedback overall, or at least feedback directly related to Pando rather than how implementation details. Because the user would have the opportunity to focus more on the actual usage of Pando, potential negative feelings associated with troubleshooting configuration just to get something to show up on the screen would be alleviated. 
+
+Another possible outcome is that time in supporting installation issues would be decreased. Hypothetically the purpose of the cli would be to remove as many steps as possible to getting an instance of Pando up and running. If the cli flow used clear language and interactive step-by-step instruction, there is less room for user error as well as more opportunity to inject personality into messaging.
+
+# Detailed design
+
+There could be multiple ways to approach making a scaffolding tool available.
+
+## An npm package
+
+The basic function of an npm package would be 
+1. Run a command
+1. Command executes a script
+1. Script clones a git project into a named directory
+
+This would allow the user to enter a command like `[npm/yarn/pnpm] create pando-design@1 [project-name]` to materialize a project structure similarly to the experience of `create-react-app`. The project structure (abbreviated) would be as follows:
```

Comment:
NPM package is preferred/current standard. Props to mentioning Yeoman but that fool is sleeping in davey jone's locker at this point. 😂  Scaffolding an entire app is overkill for a UI lib since we don't really want to dictate the entire project, so I think the route this tool would be more beneficial is probably just installing all the pando packages and setting up the normalize links in the `index.html` file `head`.

Reply:
@caseybaggz here she is: https://codesandbox.io/p/sandbox/youthful-gates-3zk7d2

Because you have to open a terminal to access it, I may have to grant your user owner or collaborator access or something. If you are not able to interact with it lmk what your username/email is for CodeSandbox and I'll try to add you.

also, I'm not sure - but you may have to `npm install -g .` in the root directory to install the package to actually run it. 

Then, these should all do something:
```
npx pando --help
npx pando setup
npx pando update
```

I never would have thought you could build a cli in CodeSandbox 😂 I learned a lot building this project. It was actually pretty fun.

**Some notes & takeaways**
* If I were to invest more time 
  * I'd make it (more) prettier (don't laugh at the grammar there. Ok you can, but I couldn't think of a better way to say it). I added some emoji's and interactive elements, but you're right - Astro's cli is pretty sweet. There are a lot of fun node packages out there.
  * There is an inherent assumption that there is an `index.html` and a `<head>` tag in that `index.html`. In future work the cli could check to ensure it is there and if not, prompt the user for a file to use. For example, in Astro all files seem to end in `.astro`, including the one with the `<head>` tag. But hypothetically as long as there's a correct path to a file with that tag, the cli should be able to add the required tags.
  * There's probably a way to extract more of the repeated code. I found that when I tried to pull functionality into shared functions outside of the script, the rest of the program would run out of order.
* CodeSandbox still makes use of CommonJs vs ES Modules. This made it a little harder to work through some of the package documentation because it all assumes ESM :) 
* There's something about writing for the terminal that is strangely satisfying from a coding perspective. Maybe it is the immediate feedback and hacker-y feeling of it all 😄 

Finally, I recorded a lil' demo just in case testing it through CodeSandbox doesn't work for some reason. Thanks again for your time in reviewing.

https://github.com/pluralsight/pando-rfcs/assets/92607316/6b5a203f-8e05-4ae1-be0c-10d9125dc022